heudiconv/tests/test_convert.py

```diff
@@ -111,3 +114,75 @@ def test_b0dwi_for_fmap(tmpdir, capfd):
         assert op.isdir(op.join(tmppath, 'sub-%s', mod) % (subID))
         for ext in ['bval', 'bvec']:
             assert glob(op.join(tmppath, 'sub-%s', mod, 'sub-%s_*.%s') % (subID, subID, ext))
+
+
+# Test two scenarios for each case:
+# -study without sessions
+# -study with sessions
+@pytest.mark.parametrize(
+    "subjects, sesID, expected_session_folder", [
+        (['Jason', 'Bourne'], None, 'sub-{sID}'),
+        (['Bourne'], 'Treadstone', op.join('sub-{{sID}}', 'ses-{{ses}}')),
+    ]
+)
+# Two possibilities: with or without heuristics:
+@pytest.mark.parametrize(
+    "heuristic", ['example', None]       # heuristics/example.py
+)
+def test_convert(tmpdir, monkeypatch, capfd,
+                 subjects, sesID, expected_session_folder,
+                 heuristic):
+    """
+    Test convert
+
+    For now, I'm just going to test that the call to populate_intended_for is
+    done with the correct argument.
+    More tests can be added here.
+    """
+
+    def mock_populate_intended_for(session, matching_parameter='Shims', criterion='Closest'):
+        """
+        Pretend we run populate_intended_for, but just print out the arguments.
+        """
+        print('session: {}'.format(session))
+        print('matching_parameter: {}'.format(matching_parameter))
+        print('criterion: {}'.format(criterion))
+        return
+    # mock the "populate_intended_for" attribute for the module from
+    # which "convert" was imported:
+    monkeypatch.setattr(
+        sys_modules[convert.__module__], "populate_intended_for", mock_populate_intended_for
+    )
```

Comment:
just `import heudiconv.convert` and  just `monkeypatch.setattr(heudiconv.convert, ...` or
`mock.patch('heudiconv.convert.populate_intended_for', new_callable=mock_populate_inteded_for)` . no need to drag sys_modules lookup into it IMHO

Reply:
I was just trying not to have to import the whole `heudiconv.convert` module, but rather only the functions being used, for consistency with other `tests/` files.
I followed your suggestion in 47bab3c8e07c818dec493b8dcfede767534c710a